pkg/resources/cruisecontrol/topicManager.go

```diff
@@ -68,6 +71,17 @@ func newCruiseControlTopic(cluster *v1beta1.KafkaCluster) *v1alpha1.KafkaTopic {
 }
 
 func generateCCTopic(cluster *v1beta1.KafkaCluster, client client.Client, log logr.Logger) error {
+	readOnlyConfigProperties := util.ParsePropertiesFormat(cluster.Spec.ReadOnlyConfig)
+
+	// for compatibility reasons the only case when we let CC to create its own kafka topics is
+	// when we enable the creation explicitly
+	if propertyValue, present := readOnlyConfigProperties[ccMetricTopicAutoCreate]; present {
+		if autoCreate, _ := strconv.ParseBool(propertyValue); autoCreate {
```

Comment:
Don't ignore the error returned by `strconv.ParseBool` but rather pass it to the caller.